tests/mocks/nrf_modem_at/mock_nrf_modem_at.c

```diff
@@ -61,28 +82,84 @@ static int scanf_mock_count;
 /* Mock information for multiple nrf_modem_at_scanf() calls */
 static struct scanf_mock scanf_mocks[CONFIG_MOCK_NRF_MODEM_AT_SCANF_CALL_COUNT];
 
+#endif
+
 void mock_nrf_modem_at_Init(void)
 {
+#if defined(CONFIG_MOCK_NRF_MODEM_AT_PRINTF)
+	memset(printf_mocks, 0, sizeof(printf_mocks));
+	printf_mock_index = 0;
+	printf_mock_count = -1;
+#endif
+#if defined(CONFIG_MOCK_NRF_MODEM_AT_SCANF)
 	memset(scanf_mocks, 0, sizeof(scanf_mocks));
 	scanf_mock_index = 0;
 	scanf_mock_count = -1;
+#endif
 }
 
 void mock_nrf_modem_at_Verify(void)
 {
+#if defined(CONFIG_MOCK_NRF_MODEM_AT_PRINTF)
+	TEST_ASSERT_MESSAGE(printf_mock_index > printf_mock_count,
+		"nrf_modem_at_printf called fewer times than expected");
+#endif
+#if defined(CONFIG_MOCK_NRF_MODEM_AT_SCANF)
 	TEST_ASSERT_MESSAGE(scanf_mock_index > scanf_mock_count,
 		"nrf_modem_at_scanf called fewer times than expected");
+#endif
+}
+
+#if defined(CONFIG_MOCK_NRF_MODEM_AT_PRINTF)
+
+void __mock_nrf_modem_at_printf_ExpectAndReturn(const char *fmt, int return_value)
+{
+	printf_mock_count++;
+
+	TEST_ASSERT_MESSAGE(
+		printf_mock_count < CONFIG_MOCK_NRF_MODEM_AT_PRINTF_CALL_COUNT,
+		"Too many nrf_modem_at_printf calls set to be expected. Please increase "
+		"CONFIG_MOCK_NRF_MODEM_AT_PRINTF_CALL_COUNT in mock_nrf_modem_at.c.");
```

Comment:
```suggestion
		"CONFIG_MOCK_NRF_MODEM_AT_PRINTF_CALL_COUNT.");
```

Reply:
Fixed. This was a leftover as there was a define in the c file earlier but I decided to make it a Kconfig option in the end.